src/JustSaying/AwsTools/MessageHandling/SnsTopicByName.cs

```diff
@@ -85,49 +85,49 @@ public override async Task<bool> ExistsAsync()
             return false;
         }
 
-        public async Task<bool> CreateAsync()
+        public async Task CreateAsync()
         {
             try
             {
                 var response = await Client.CreateTopicAsync(new CreateTopicRequest(TopicName))
                     .ConfigureAwait(false);
 
-                if (!string.IsNullOrEmpty(response.TopicArn))
+                if (string.IsNullOrEmpty(response.TopicArn))
                 {
-                    Arn = response.TopicArn;
-                    _log.LogInformation("Created topic '{TopicName}' on ARN '{Arn}'.", TopicName, Arn);
-                    return true;
+                    _log.LogError("Failed to create topic or obtain ARN for topic {TopicName}", TopicName);
+                    throw new Exception($"Failed to create topic or receive ARN '{TopicName}'.");
                 }
-                _log.LogWarning("Failed to create topic '{TopicName}'.", TopicName);
+
+                Arn = response.TopicArn;
+                _log.LogInformation("Created topic '{TopicName}' on ARN '{Arn}'.", TopicName, Arn);
             }
             catch (AuthorizationErrorException ex)
             {
                 _log.LogWarning(0, ex, "Not authorized to create topic '{TopicName}'.", TopicName);
                 if (!await ExistsAsync().ConfigureAwait(false))
                 {
-                    throw new InvalidOperationException($"Topic '{TopicName}' does not exist, and no permission to create it.");
+                    throw new InvalidOperationException(
+                        $"Topic '{TopicName}' does not exist, and no permission to create it.");
                 }
             }
-
-            return false;
         }
 
-        public async Task<bool> CreateWithEncryptionAsync(ServerSideEncryption config)
+        public async Task CreateWithEncryptionAsync(ServerSideEncryption config)
         {
-            var created = await CreateAsync().ConfigureAwait(false);
+            await CreateAsync().ConfigureAwait(false);
 
-            ServerSideEncryption = await ExtractServerSideEncryptionFromTopicAttributes().ConfigureAwait(false);
+            ServerSideEncryption =
+                await ExtractServerSideEncryptionFromTopicAttributes().ConfigureAwait(false);
 
             await EnsureServerSideEncryptionIsUpdatedAsync(config).ConfigureAwait(false);
-
-            return created;
         }
 
         private async Task<ServerSideEncryption> ExtractServerSideEncryptionFromTopicAttributes()
         {
             var attributesResponse = await Client.GetTopicAttributesAsync(Arn).ConfigureAwait(false);
 
-            if (!attributesResponse.Attributes.TryGetValue(JustSayingConstants.AttributeEncryptionKeyId, out var encryptionKeyId))
+            if (!attributesResponse.Attributes.TryGetValue(JustSayingConstants.AttributeEncryptionKeyId,
```

Comment:
```suggestion
            if (!attributesResponse.Attributes.TryGetValue(
                JustSayingConstants.AttributeEncryptionKeyId,
```

Reply:
Not sure this one makes sense - I'm formatting with the editorconfig settings and this line doesn't pass the char limit.

Reply:
Maybe just remove the "length" rule. It's causing so much code churn and making it less readable IMHO.

Reply:
I think the churn is because we have many files that haven't been formatted before, so as we touch them and format them as they change. 

We could possibly do a global format, though it would likely invalidate Stu's other PR.

I think we're going to get churn with auto formatting rules regardless until everything is formatted with the ruleset. Personally I feel that auto formatting is a better solution than ad-hoc fixes in every PR.

Reply:
If it were, I wouldn't auto-format the files and only commit the changes I'm intentionally making.

Reply:
I'm curious what your workflow is, as I'm keen to find a solution here! Do you use auto-format? If so is it just for the code you change? What if it's surrounded by unformatted code? 

If we were in a utopia where everything was already formatted, then it wouldn't really be a problem as the only changes that would exist after an auto-format would be the intentional changes.

Can we get there somehow? Do we want to?

Reply:
Visual Studio doesn't appear to be super-aggressive with formatting the whole document you're working on (VS Code is, for example) unless you specifically ask it to, and that's what I use. I don't have ReSharper either, I've never liked it.

Then I use GitHub Desktop as my Git client, and I manually deselect (or undo in the GHD UI) lines that aren't anything to do with what I'm trying to change.

I might bring a handful of fix-ups with me, but I won't generally do wholesale changes when others have to review it, as it just adds extra friction to the process of getting the PR over the line for merging. Otherwise you can end-up with a snowball effect where a few simple changes end up as a giant PR, with each bit of feedback being addressed changing more and more formatting.

If we want to make every single file be 100% compliant with the EditorConfig file, then that can be done separately as a dedicated piece of work if that's something we feel is worth doing.

JustSaying is 6+ years old and I'm not sure churning the whole codebase to hit an arbitrary formatting Nirvana is worth the time and investment given its used less these days with many workers being replaced with Lambdas with SQS triggers.

For my 2 cents, the EditorConfig is just there as a preference on entirely new code and edits to existing code, rather than a set of rules that must be obeyed. If we wanted formatting _rules_ I'd opt for StyleCop (which I use in my own projects).

Reply:
> Not sure this one makes sense - I'm formatting with the editorconfig settings and this line doesn't pass the char limit.

I simply suggested the change because on line 162 you put all the parameters on a separate line, so, for consistency, I thought you should do the same on this too. I didn't want to create a long thread about formatting conventions 😅

Reply:
Thanks @martincostello, I appreciate your perspective. I'll avoid changing formatting going forward, and maybe consider doing a global format but as you say, there might not be a huge benefit. I tend to compulsively format on save in rider, so it can be a challenge to contribute to code bases without auto format rules setup 😅.

Thanks for your time and patience!

Reply:
As for this specific comment, I've moved the parameter to the next line, so will resolve.